src/redux/actions/messages.js

```diff
@@ -0,0 +1,215 @@
+import { createAction } from 'redux-actions';
+import actionTypes from '../actionTypes';
+import _ from 'lodash';
+import * as apis from '../../apis';
+import store from '../../store';
+
+const returnArgumentsFn = function(payload) {
+  return Promise.resolve(payload);
+};
+
+const messageActions = {
+  markAllRead: createAction(actionTypes.MESSAGES_MARK_ALL_READ),
+  createMessage: createAction(
+    actionTypes.MESSAGES_CREATE_USER_MESSAGE,
+    payload => {
```

Comment:
Do not do this part here. Use `returnArgumentsFn` here.

Reply:
Sir, I was planning to use ```returnArgumentsFn``` but then I realized that I have to call other actions from within thin action, so I went for this implementation. Where ever I have not used ```returnArgumentsFn``` it is because I had to either call another action or modify the payload before passing it to the reducer. Let me know if there is a more elegant way of doing this that I might be missing out. Thankyou for your time, sir. @akshatnitd

Reply:
There is no issue in calling another action from within another in a component.

Reply:
```
createMessage: createAction(
    actionTypes.MESSAGES_CREATE_USER_MESSAGE,
    payload => {
      const timestamp = Date.now();
      const { text, voice } = payload;
      payload.message = {
        id: 'm_' + timestamp,
        authorName: 'You',
        date: new Date(timestamp),
        text: text,
        type: 'message',
        voice: voice,
      };
      apis
        .getSusiReply(payload)
        .then(response => {
          console.log(response);
          store.dispatch(messageActions.createSusiMessage({ response, voice }));
        })
        .catch(error => {
          console.log(error);
        });
      return Promise.resolve(payload);
    },
  ),
```

Sir for example in this code, I am manipulating the payload before passing it on to the api, in this case, if I call the action/ manipulate the payload from the component where this action is called, than the manipulation code and the second action call would need to be repeated every time this action would be called in a component. I hope I haven't got this wrong in my head. Thank you for your detailed review. @akshatnitd

Reply:
calling just another action would have been easy to be shifted to the components, but payload manipulation, not so sure.

Reply:
There is not issue for the manipulation to be done in the component itself. It will give the developer more sense over the actions while being called. For making the component code cleaner, you should make a utils for it ( eg - `formatUserMessage`). And then call another action with the formatted message.

Reply:
I see that the formatting for user message and SUSI message is used at multiple places, so would be good to make utils for them.

Reply:
@akshatnitd Okay this, I think the new commit has addressed the requested changes. I have got rid of the repetition in the message formatting utility file. I have also removed the payload modification and api calls from actions like you asked.